user_tools/src/spark_rapids_pytools/common/sys_storage.py

```diff
@@ -239,7 +239,7 @@ def gen_dir_tree(cls,
         dir_count = 0
 
         def inner(dir_p: pathlib.Path, prefix: str = '', level=-1):
-            nonlocal files_count, dir_count, dir_patterns, file_patterns
+            nonlocal files_count, dir_count
```

Comment:
I see `dir_patterns` and `file_patterns` are being used.
I guess it is because it is nort needed to be redefined as `non_local`.

Would this work for python 3.9+?

Reply:
That is correct. Since these are read-only variables (defined in the outer function), we do not need them to be `non_local`.

Yes, all tox tests passed for python versions 3.9-3.12.

Reply:
Yes. That is true. In this case, I think it should be fine because these variables are read-only and these variables can be safely read in the inner function.